server/src/types/resolvers/Channel/mutation.ts

```diff
@@ -135,35 +119,34 @@ export const createChannel = mutationField('createChannel', {
 
         const totalUsers = userIds.length + 1; // +1 for auth user
 
-        let existingChannel: Channel;
+        return channels.find(channel => channel.membership.length === totalUsers)
```

Comment:
How about adding
```
cas hasSameMembershipLength = channel.membership.length === totalUsers
```

Reply:
In my opinion, it would be nice to do something like this:
`const findChannelWithUserIds = async (userId, userIds) => {
    return await ctx.prisma.channel.findMany({
    ...
 });
 const findChannelWithMembershipLength = (channels, length) => {
   return channels.find((channel) => channel.membership.length == length)
 };
 ...
 const userChannels = await findChannelWithUserIds(userId, userIds);
 const totalUsers = userIds.length + 1; // +1 for auth user
 const existingChannel = findChannelWithMembershipLength(userChannels, totalUsers)
`

The function findChannelWithUserIds currently does two things. An action to find a channel and an action to filter a channel that meets a specific condition within the channel. How about splitting these two moves?

Reply:
In my opinion, it would be nice to do something like

The following function immediately returns the result of executing the query.
> const findChannelWithUserIds = async (userId, userIds) => {
    return await ctx.prisma.channel.findMany({
    ...
});


Declare a function to find channels with membership length in the channel list.
> const findChannelWithMembershipLength = (channels, length) => {
  return channels.find((channel) => channel.membership.length == length)
};
...

The main logic can be organized as follows.
> const userChannels = await findChannelWithUserIds(userId, userIds);
> const totalUsers = userIds.length + 1; // +1 for auth user
> const existingChannel = findChannelWithMembershipLength(userChannels, totalUsers);

The function "findChannelWithUserIds" currently does two things. The operation of searching for a channel as above and filtering the channel that meets a specific condition within the channel. The currently defined function name does not describe both of these actions, and it is ideal that the function performs one action as much as possible, so how about dividing these two actions as above?

Reply:
> Actually, I think inlining everything is the way to go. We're not reusing these functions, so there is no need to make more named functions. I suggest doing the following:
> 
> ```ts
> const userChannels = await ctx.prisma.channel.findMany({
>   // ...
> });
> const totalUsers = userIds.length + 1; // +1 for auth user.
> const existingChannel = userChannels.find(channel => channel.membership.length === totalUsers);
> ```

It is considered that this part requires more discussion than expected. In the above code, all functions written below `resolve`(line 45) are not likely to be reused. Nevertheless, I think that it is declared and used as a function to create more readable code. 
In this case, `findChannelWithUserIds`, a function that returns `UserChannels`, is declared as a function to improve readability, and after that, I think it is better to do the same as you said. 
What do you think?

Reply:
> I half agree with this decision (well but very nice to think about this). `findChannelWithUserIds` actually doesn't do what's expected since it does not return actual `channel` since we need to filter its length.
> 
> Therefore, it is pretty awkward to separate internal behaviors and make them into two functions since `findChannelWithUserIds` itself does not resolve what's expected.
> 
> How do you think about my point of view?

I think this is a problem that can vary depending on individual perspectives and styles. In my opinion, `findUsers` returns a list of Users, and if it returns the length, `findUsersCnt` is suitable.

Reply:
Well to saturate our discussion I think we need 1 function with two nested ones as you have mentioned. Our main goal is to get unique channel but then we need to work on others to make it happen.